tests/core/test_scripts.py

```diff
@@ -119,3 +122,183 @@ def test_delete_collection_raise_if_the_collection_does_not_exist(self):
+            object_id='collection_info',
+            record={'record_count': 1, 'storage_size': 160})
+
+        mocked.info.assert_any_call('Bucket bucket-1, collection collection-1. Final size: 1 records, 78 bytes.')
+        mocked.info.assert_any_call('Bucket bucket-1, collection collection-2. Final size: 1 records, 79 bytes.')
+        mocked.info.assert_any_call('Bucket bucket-1. Final size: 2 records, 193 bytes.')
+
+    def test_rebuild_quotas_doesnt_update_if_dry_run(self):
+        get_all_data = [
+            ([{"id": "bucket-1", "last_modified": 10}], 1),   # get buckets
+            ([{"id": "collection-1", "last_modified": 100}], 1),   # get collections for first bucket
+            # get records for first collection
+            ([{"id": "record-1", "last_modified": 110}], 1),
+            ([], 0),
+
+            # trying to get remaining page of collections
+            ([], 0),
+
+            # getting remaining page of buckets
+            ([], 0),
+        ]
+        self.registry.storage.get_all.side_effect = lambda *args, **kwargs: get_all_data.pop(0)
+
+        with mock.patch('kinto.core.scripts.logger') as mocked:
+            scripts.rebuild_quotas({'registry': self.registry})
+
+        self.registry.storage.get_all.assert_any_call(
+            collection_id='bucket',
+            limit=self.BATCH_SIZE,
+            parent_id='',
+            pagination_rules=None,
+            sorting=[self.OLDEST_FIRST],
+            )
+        self.registry.storage.get_all.assert_any_call(
+            collection_id='collection',
+            limit=self.BATCH_SIZE,
+            parent_id='/buckets/bucket-1',
+            pagination_rules=None,
+            sorting=[self.OLDEST_FIRST],
+            )
+        self.registry.storage.get_all.assert_any_call(
+            collection_id='record',
+            limit=self.BATCH_SIZE,
+            parent_id='/buckets/bucket-1/collections/collection-1',
+            pagination_rules=None,
+            sorting=[self.OLDEST_FIRST],
+            )
+        self.registry.storage.get_all.assert_any_call(
+            collection_id='record',
+            limit=self.BATCH_SIZE,
+            parent_id='/buckets/bucket-1/collections/collection-1',
+            pagination_rules=[[Filter('last_modified', 110, COMPARISON.GT)]],
+            sorting=[self.OLDEST_FIRST],
+            )
+        self.registry.storage.get_all.assert_any_call(
+            collection_id='collection',
+            limit=self.BATCH_SIZE,
+            parent_id='/buckets/bucket-1',
+            pagination_rules=[[Filter('last_modified', 100, COMPARISON.GT)]],
+            sorting=[self.OLDEST_FIRST],
+            )
+        self.registry.storage.get_all.assert_any_call(
+            collection_id='bucket',
+            limit=self.BATCH_SIZE,
+            parent_id='',
+            pagination_rules=[[Filter('last_modified', 10, COMPARISON.GT)]],
+            sorting=[self.OLDEST_FIRST],
+            )
```

Comment:
Maybe in this one we can just assert about the output and not individual calls to `get_all()` ?